spec/message.abnf

```diff
@@ -1,43 +1,46 @@
-message = pattern / complex-message
+message = simple-message / complex-message
 
-complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
+simple-message = [simple-start pattern]
+simple-start = simple-start-char / text-escape / expression
+pattern = *(text-char / text-escape / expression)
```

Comment:
I'm unhappy about losing the `text` production. Conceptually, we think of patterns as sequences of contiguous text runs or placeholders, not a sequence of characters or placeholders.

Perhaps we shoud have `quoted-text` and `unquoted-text`? The latter would replace `simple-message`.

Reply:
I agree. This was better as `*(text / expression)`

Reply:
Note that `simple-start` is only applied to the first character, and it needs to be a part of the eventual _text_. Furthermore, a simple message pattern may include multiple _text_ spans separated by expressions.

To see this, consider the valid message `one{$x}.two`. How could you express a `text` that captures both `one` and `.two`, but would not if their order was reversed as `.two{$x}one`, which is invalid?

Reply:
This seems like a request for what I labeled "Data Model Analog" in https://github.com/unicode-org/message-format-wg/pull/529#discussion_r1401328223 .
> ```abnf
> simple-message    = [simple-start-part *pattern-part]
> simple-start-part = simple-start-text / expression
> …
> ; pattern is a possibly-empty sequence of text and expression parts
> pattern      = *pattern-part
> pattern-part = text / expression
> ```

Reply:
During today's call we resolved to be satisfied with _text_ being well defined in `syntax.md`, and not having a single `text` ABNF rule.